csm/core/agent/csm_agent.py

```diff
@@ -106,7 +106,8 @@ def init():
         auth_service = AuthService()
         user_manager = UserManager(db)
         role_manager = RoleManager(roles)
-        session_manager = SessionManager()
+        session_manager = SessionManager(Conf.get(const.CSM_GLOBAL_INDEX, const.SESSION_BACKEND_KEY),
```

Comment:
This should be backward compatible. No need to pass config in constructor. 

Make db parameter as optional .

In the service always have a check for NULL value for db

If you do not get config please keep the default to 'persistent'

Reply:
db parameter is optional--> I will add check for null
 I am not passing config in constructor --> I am reading session_backend key word from conf and passing it here
Conf.get(const.CSM_GLOBAL_INDEX, const.SESSION_BACKEND_KEY)

Reply:
updated